theano/tensor/tests/test_opt.py

```diff
@@ -3547,6 +3548,34 @@ def test2(self):
 
         self._verify_stack_trace(f)
 
+    def test_useless_alloc_with_shape_one(self):
+        alloc_lift = out2in(local_useless_alloc)
+        x = shared(self.rng.randn(2,))
+        y = shared(self.rng.randn())
+        alloc_x = tensor.alloc(x, 1, 3, 2)
+        alloc_y = tensor.alloc(y, 1, 1)
+
+        g = FunctionGraph([x, y], [alloc_x, alloc_y])
+        self.assertTrue(str(g) == ("[Alloc(<TensorType(float64, vector)>, "
+                                   "TensorConstant{1}, "
+                                   "TensorConstant{3}, "
+                                   "TensorConstant{2}), "
+                                   "Alloc(<TensorType(float64, scalar)>, "
+                                   "TensorConstant{1}, "
+                                   "TensorConstant{1})]"))
+
+        alloc_lift.optimize(g)
+        self.assertTrue(str(g) == "[DimShuffle{x,0,1}"
+                                  "(Alloc(<TensorType(float64, vector)>, "
+                                  "TensorConstant{3}, "
+                                  "TensorConstant{2})), "
+                                  "DimShuffle{x,x}"
+                                  "(<TensorType(float64, scalar)>)]")
+
+        # Check stacktrace was copied over correctly after opt was applied
+        self.assertTrue(hasattr(g.outputs[0].tag, 'trace'))
```

Comment:
Actually, `self._verify_stack_trace` has been removed since all of the other tests removed nodes, and there was no stack trace to check.
We have upgraded the way of checking for traces as well, we have a function for that, [`check_stack_trace`](https://github.com/Theano/Theano/blob/3c70348fd7c31fe169cd2038999e3a5ba106d738/theano/gof/opt.py#L2672) that accepts either functions or fgraphs.
So, you can simply call:
```python
self.assertTrue(check_stack_trace(g, ops_to_check='all'))
```
If it gives an error, please report it and we will check if the optimization has to be updated, or if the check should be less strict.